examples/decorator/noxfile.py

```diff
@@ -0,0 +1,8 @@
+import nox
+
+
+@nox.session
+def python(session):
+    session.install("-rrequirements-dev.txt")
+    session.install(".")
```

Comment:
To avoid having yet another environment generated for the pep 517 build

```suggestion
    session.install("-e", ".", "--no-build-isolation")
```

... and similar for the other examples.